OrbitSsh/LocalSocketManager.cpp

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitSsh/LocalSocketManager.h"
+
+#include "OrbitBase/Logging.h"
+#include "OrbitSsh/ResultType.h"
+
+namespace OrbitSsh {
+
+LocalSocketManager::LocalSocketManager(std::string address, int port)
+    : address_(address), port_(port) {}
+
+ResultType LocalSocketManager::Accept() {
+  if (!listen_socket_->CanBeRead()) return ResultType::kAgain;
+
+  std::unique_ptr<Socket> new_socket;
+  ResultType result = listen_socket_->Accept(&new_socket);
+
+  if (result == ResultType::kSuccess) {
+    accepted_socket_ = std::move(new_socket);
+  }
+
+  return result;
+}
+
+// Tick manages the current state and progresses when appropriate.
+// * kNotInitialized means both listen_socket and accepted_socket are not
+// initialized aka nullptr. In this state it is tried to create a listen_socket.
+// * kInitialized means the listen_socket exists and it is tried to bind it to
+// the address and port
+// * kBound means listen_socket bind was successful and the next step is start
+// listening
+// * kListening means the listen_socket is listening until a connection on the
+// socket can be accepted
+// * kRunning means there is an active connection on the accepted_socket.
+// * kShutdownSent means a shutdown signal has been sent and the
+// accepted_channel waits on a disconnect signal to arrive.
+// * kRemoteDisconnect means there is no active connection on the accepted
+// socket anymore. The socket will be deleted and the state goes back to
+// listening for new connections (kListening)
```

Comment:
can you move these to header file? Ideally documentation should go before class declaration.